### ISSUE TITLE
libevent_pthreads requires libevent not libevent_core

### ISSUE THREAD

**Issue by moonlightsh** [libevent/libevent#1213]
Hello,
When I use `libevent_pthreads` by `pkgconfig`, 
```
$ pkg-config --libs libevent_pthreads
-L/usr/local/lib -levent_pthreads -levent
```
But  [here](http://www.wangafu.net/~nickm/libevent-book/Ref0_meta.html) mentioned 

>libevent
This library exists for historical reasons; it contains the contents of both libevent_core and libevent_extra. You shouldn’t use it; it may go away in a future version of Libevent.

So , I put `event_core` and `event_pthreads`  in the system, no `event`. But linker will abort because of `libevent not found`, I have to add link flags manually.
And then ,I check the `libevent_pthreads.pc`,  `Requires: libevent` was defined.  I don't kown if there are any improvements here.

https://github.com/libevent/libevent/blob/048907a501e2ee106cd9dabf6686697f3b73b562/libevent_pthreads.pc.in#L1-L15

**Comment by azat** [comment]
Hi!

What version of libevent do you have?
The issue that you got had been fixed in 731469b33a683b359783b4658994081971e8f474

**Comment by moonlightsh** [comment]
Thks,
I checked https://github.com/libevent/libevent/commit/731469b33a683b359783b4658994081971e8f474 ,  I found this commit is about `libevent.pc`. 
And at the master branch of libevent, if we use `pkg-config` to get the dependences of `libevent_pthreads` , it shows `-L/usr/local/lib -levent_pthreads -levent`. So it will add `-levent` automatically. Then we have to leave a `libevent.so` there.

**Comment by azat** [comment]
>And at the master branch of libevent, if we use pkg-config to get the dependences of libevent_pthreads , it shows -L/usr/local/lib -levent_pthreads -levent. So it will add -levent automatically. Then we have to leave a libevent.so there.

I can't confirm this.
Can you provide a reproducer? in docker

**Comment by azat** [comment]
```sh
# docker run --rm --name le -it ubuntu:20.04 bash
# apt-get update
# apt install git gcc automake autoconf libtool pkg-config make --no-install-recommends
# git clone git://github.com/libevent/libevent.git
# cd libevent
# ./autogen.sh
# ./configure --disable-openssl --disable-dependency-tracking
# make -j install
# pkg-config --libs libevent_pthreads
-L/usr/local/lib -levent_pthreads -levent_core -levent_extra
```

**Comment by ploxiln** [comment]
@moonlightsh must be using the latest release 2.1.12 (or older), the behavior is changed between the master branch and the patches-2.1 branch.

(Should libevent_pthreads depend on libevent_extra btw?)

------------------

```
docker run --rm -it debian:11 /bin/bash
```

inside run:

```
apt update
apt install build-essential pkg-config wget
cd /tmp
wget https://github.com/libevent/libevent/releases/download/release-2.1.12-stable/libevent-2.1.12-stable.tar.gz
./configure --disable-openssl
make
make install
pkg-config --libs libevent_pthreads
```

final output:

```
-L/usr/local/lib -levent_pthreads -levent
```

**Comment by azat** [comment]
>(Should libevent_pthreads depend on libevent_extra btw?)

@ploxiln Indeed, I've looked thought and seems that you are right.
Can you submit a patch for this?

**Comment by azat** [comment]
This issue can be closed, since the initial problem had been fixed in upstream/master.

**Comment by moonlightsh** [comment]
@azat  Now, I checkout the master branch and test it again. The previous description isn't clearly:
  Only pkg `libevent_core` and `libevent_pthread` in my project, in other words, I do't need `libevent.so` nor `libevent_extra.so`. And I think `libevent_pthreads` and `libevent_openssl` no need to require `libevent` nor `libevent_extra`.  I try to fix at [#2539c8a3ebd6a9b44bc05dd9ad5dfd627efd644e](https://github.com/moonlightsh/libevent/commit/2539c8a3ebd6a9b44bc05dd9ad5dfd627efd644e), can u help me to review it

**Comment by azat** [comment]
>LGTM, can you submit it as MR (to run CI) ?

Thank you for the effort, I've submitted MR on behalf of you.
